Content.Server/Holiday/Christmas/GiftPackinSystem.cs

```diff
@@ -0,0 +1,102 @@
+﻿using Content.Server.Hands.Systems;
+using Content.Shared.Examine;
+using Content.Shared.Interaction.Events;
+using Content.Shared.Item;
+using Robust.Server.GameObjects;
+using Robust.Shared.Map.Components;
+using Robust.Shared.Physics.Components;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+
+namespace Content.Server.Holiday.Christmas;
+
+/// <summary>
+/// This handles granting players their gift.
+/// </summary>
+public sealed class GiftPackinSystem : EntitySystem
+{
+    [Dependency] private readonly AudioSystem _audio = default!;
+    [Dependency] private readonly HandsSystem _hands = default!;
+    [Dependency] private readonly IComponentFactory _componentFactory = default!;
+    [Dependency] private readonly IPrototypeManager _prototype = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+
+    private readonly List<string> _possibleGiftsSafe = new();
+    private readonly List<string> _possibleGiftsUnsafe = new();
+
+    /// <inheritdoc/>
+    public override void Initialize()
+    {
+        _prototype.PrototypesReloaded += OnPrototypesReloaded;
+        SubscribeLocalEvent<GiftPackinComponent, MapInitEvent>(OnGiftMapInit);
+        SubscribeLocalEvent<GiftPackinComponent, UseInHandEvent>(OnUseInHand);
+        SubscribeLocalEvent<GiftPackinComponent, ExaminedEvent>(OnExamined);
+        BuildIndex();
+    }
+
+    private void OnExamined(EntityUid uid, GiftPackinComponent component, ExaminedEvent args)
+    {
+        if (!component.ContentsViewers.IsValid(args.Examiner, EntityManager) || component.SelectedEntity is null)
+            return;
+
+        var name = _prototype.Index<EntityPrototype>(component.SelectedEntity).Name;
+        args.Message.PushNewline();
+        args.Message.AddText(Loc.GetString("gift-packin-contains", ("name", name)));
+    }
+
+    private void OnUseInHand(EntityUid uid, GiftPackinComponent component, UseInHandEvent args)
+    {
+        if (args.Handled)
+            return;
+
+        if (component.SelectedEntity is null)
+            return;
+
+        var coords = Transform(args.User).Coordinates;
+        var handsEnt = Spawn(component.SelectedEntity, coords);
+        EnsureComp<ItemComponent>(handsEnt); // For insane mode.
+        if (component.Wrapper is not null)
+            Spawn(component.Wrapper, coords);
+
+        args.Handled = true;
+        _audio.PlayPvs(component.Sound, args.User);
+        Del(uid);
```

Comment:
Should probably QueueDel just in case this breaks some interaction logic somewhere

Reply:
Can't queuedel as then it's blocking the user's hands. The event gets handled, if it breaks interaction logic it's the fault of the handler not checking for handled, and not this code.